Test failures with PROJ 9.0.1

### Description
The first release candidates for PROJ 9.0.1 have been released, the release contains changes that cause test failures due to almost_equal tests failing.

```

=================================== FAILURES ===================================
______________________________ test_default[land] ______________________________

emphasis = 'land'

    @pytest.mark.parametrize("emphasis", ["land", "ocean"])
    def test_default(emphasis):
        igh = ccrs.InterruptedGoodeHomolosine(emphasis=emphasis)
        other_args = {"ellps=WGS84", "lon_0=0"}
        if emphasis == "land":
            check_proj_params("igh", igh, other_args)
        elif emphasis == "ocean":
            check_proj_params("igh_o", igh, other_args)
        assert_almost_equal(
            np.array(igh.x_limits), [-20037508.3427892, 20037508.3427892]
        )
>       assert_almost_equal(
            np.array(igh.y_limits), [-8683259.7164347, 8683259.7164347]
        )
E       AssertionError: 
E       Arrays are not almost equal to 7 decimals
E       
E       Mismatched elements: 2 / 2 (100%)
E       Max absolute difference: 0.00017592
E       Max relative difference: 2.02598163e-11
E        x: array([-8683259.7162588,  8683259.7162588])
E        y: array([-8683259.7164347,  8683259.7164347])

cartopy/tests/crs/test_interrupted_goode_homolosine.py:30: AssertionError
_____________________________ test_default[ocean] ______________________________

emphasis = 'ocean'

    @pytest.mark.parametrize("emphasis", ["land", "ocean"])
    def test_default(emphasis):
        igh = ccrs.InterruptedGoodeHomolosine(emphasis=emphasis)
        other_args = {"ellps=WGS84", "lon_0=0"}
        if emphasis == "land":
            check_proj_params("igh", igh, other_args)
        elif emphasis == "ocean":
            check_proj_params("igh_o", igh, other_args)
        assert_almost_equal(
            np.array(igh.x_limits), [-20037508.3427892, 20037508.3427892]
        )
>       assert_almost_equal(
            np.array(igh.y_limits), [-8683259.7164347, 8683259.7164347]
        )
E       AssertionError: 
E       Arrays are not almost equal to 7 decimals
E       
E       Mismatched elements: 2 / 2 (100%)
E       Max absolute difference: 0.00017592
E       Max relative difference: 2.02598163e-11
E        x: array([-8683259.7162588,  8683259.7162588])
E        y: array([-8683259.7164347,  8683259.7164347])

cartopy/tests/crs/test_interrupted_goode_homolosine.py:30: AssertionError
______________________ test_central_longitude[land--10.0] ______________________

emphasis = 'land', lon = -10.0

    @pytest.mark.parametrize(
        ("emphasis", "lon"),
        [("land", -10.0), ("land", 10.0), ("ocean", -10.0), ("ocean", 10.0)],
    )
    def test_central_longitude(emphasis, lon):
        igh = ccrs.InterruptedGoodeHomolosine(
            central_longitude=lon, emphasis=emphasis
        )
        other_args = {"ellps=WGS84", f"lon_0={lon}"}
        if emphasis == "land":
            check_proj_params("igh", igh, other_args)
        elif emphasis == "ocean":
            check_proj_params("igh_o", igh, other_args)
    
        assert_almost_equal(
            np.array(igh.x_limits),
            [-20037508.3427892, 20037508.3427892],
            decimal=5,
        )
>       assert_almost_equal(
            np.array(igh.y_limits), [-8683259.7164347, 8683259.7164347]
        )
E       AssertionError: 
E       Arrays are not almost equal to 7 decimals
E       
E       Mismatched elements: 2 / 2 (100%)
E       Max absolute difference: 0.00017592
E       Max relative difference: 2.02598163e-11
E        x: array([-8683259.7162588,  8683259.7162588])
E        y: array([-8683259.7164347,  8683259.7164347])

cartopy/tests/crs/test_interrupted_goode_homolosine.py:68: AssertionError
______________________ test_central_longitude[land-10.0] _______________________

emphasis = 'land', lon = 10.0

    @pytest.mark.parametrize(
        ("emphasis", "lon"),
        [("land", -10.0), ("land", 10.0), ("ocean", -10.0), ("ocean", 10.0)],
    )
    def test_central_longitude(emphasis, lon):
        igh = ccrs.InterruptedGoodeHomolosine(
            central_longitude=lon, emphasis=emphasis
        )
        other_args = {"ellps=WGS84", f"lon_0={lon}"}
        if emphasis == "land":
            check_proj_params("igh", igh, other_args)
        elif emphasis == "ocean":
            check_proj_params("igh_o", igh, other_args)
    
        assert_almost_equal(
            np.array(igh.x_limits),
            [-20037508.3427892, 20037508.3427892],
            decimal=5,
        )
>       assert_almost_equal(
            np.array(igh.y_limits), [-8683259.7164347, 8683259.7164347]
        )
E       AssertionError: 
E       Arrays are not almost equal to 7 decimals
E       
E       Mismatched elements: 2 / 2 (100%)
E       Max absolute difference: 0.00017592
E       Max relative difference: 2.02598163e-11
E        x: array([-8683259.7162588,  8683259.7162588])
E        y: array([-8683259.7164347,  8683259.7164347])

cartopy/tests/crs/test_interrupted_goode_homolosine.py:68: AssertionError
_____________________ test_central_longitude[ocean--10.0] ______________________

emphasis = 'ocean', lon = -10.0

    @pytest.mark.parametrize(
        ("emphasis", "lon"),
        [("land", -10.0), ("land", 10.0), ("ocean", -10.0), ("ocean", 10.0)],
    )
    def test_central_longitude(emphasis, lon):
        igh = ccrs.InterruptedGoodeHomolosine(
            central_longitude=lon, emphasis=emphasis
        )
        other_args = {"ellps=WGS84", f"lon_0={lon}"}
        if emphasis == "land":
            check_proj_params("igh", igh, other_args)
        elif emphasis == "ocean":
            check_proj_params("igh_o", igh, other_args)
    
        assert_almost_equal(
            np.array(igh.x_limits),
            [-20037508.3427892, 20037508.3427892],
            decimal=5,
        )
>       assert_almost_equal(
            np.array(igh.y_limits), [-8683259.7164347, 8683259.7164347]
        )
E       AssertionError: 
E       Arrays are not almost equal to 7 decimals
E       
E       Mismatched elements: 2 / 2 (100%)
E       Max absolute difference: 0.00017592
E       Max relative difference: 2.02598163e-11
E        x: array([-8683259.7162588,  8683259.7162588])
E        y: array([-8683259.7164347,  8683259.7164347])

cartopy/tests/crs/test_interrupted_goode_homolosine.py:68: AssertionError
______________________ test_central_longitude[ocean-10.0] ______________________

emphasis = 'ocean', lon = 10.0

    @pytest.mark.parametrize(
        ("emphasis", "lon"),
        [("land", -10.0), ("land", 10.0), ("ocean", -10.0), ("ocean", 10.0)],
    )
    def test_central_longitude(emphasis, lon):
        igh = ccrs.InterruptedGoodeHomolosine(
            central_longitude=lon, emphasis=emphasis
        )
        other_args = {"ellps=WGS84", f"lon_0={lon}"}
        if emphasis == "land":
            check_proj_params("igh", igh, other_args)
        elif emphasis == "ocean":
            check_proj_params("igh_o", igh, other_args)
    
        assert_almost_equal(
            np.array(igh.x_limits),
            [-20037508.3427892, 20037508.3427892],
            decimal=5,
        )
>       assert_almost_equal(
            np.array(igh.y_limits), [-8683259.7164347, 8683259.7164347]
        )
E       AssertionError: 
E       Arrays are not almost equal to 7 decimals
E       
E       Mismatched elements: 2 / 2 (100%)
E       Max absolute difference: 0.00017592
E       Max relative difference: 2.02598163e-11
E        x: array([-8683259.7162588,  8683259.7162588])
E        y: array([-8683259.7164347,  8683259.7164347])

cartopy/tests/crs/test_interrupted_goode_homolosine.py:68: AssertionError
_________________________________ test_default _________________________________

    def test_default():
        moll = ccrs.Mollweide()
        other_args = {'a=6378137.0', 'lon_0=0'}
        check_proj_params('moll', moll, other_args)
    
        assert_almost_equal(np.array(moll.x_limits),
                            [-18040095.6961473, 18040095.6961473])
>       assert_almost_equal(np.array(moll.y_limits),
                            [-9020047.8480736, 9020047.8480736])
E       AssertionError: 
E       Arrays are not almost equal to 7 decimals
E       
E       Mismatched elements: 2 / 2 (100%)
E       Max absolute difference: 0.00017584
E       Max relative difference: 1.94946876e-11
E        x: array([-9020047.8478978,  9020047.8478978])
E        y: array([-9020047.8480736,  9020047.8480736])

cartopy/tests/crs/test_mollweide.py:26: AssertionError
______________________________ test_ellipse_globe ______________________________

    def test_ellipse_globe():
        globe = ccrs.Globe(ellipse='WGS84')
        with pytest.warns(UserWarning,
                          match='does not handle elliptical globes.') as w:
            moll = ccrs.Mollweide(globe=globe)
            assert len(w) == 1
    
        other_args = {'ellps=WGS84', 'lon_0=0'}
        check_proj_params('moll', moll, other_args)
    
        # Limits are the same as default since ellipses are not supported.
        assert_almost_equal(moll.x_limits, [-18040095.6961473, 18040095.6961473])
>       assert_almost_equal(moll.y_limits, [-9020047.8480736, 9020047.8480736])
E       AssertionError: 
E       Arrays are not almost equal to 7 decimals
E       
E       Mismatched elements: 2 / 2 (100%)
E       Max absolute difference: 0.00017584
E       Max relative difference: 1.94946876e-11
E        x: array([-9020047.8478978,  9020047.8478978])
E        y: array([-9020047.8480736,  9020047.8480736])

cartopy/tests/crs/test_mollweide.py:52: AssertionError
________________________ test_central_longitude[-10.0] _________________________

lon = -10.0

    @pytest.mark.parametrize('lon', [-10.0, 10.0])
    def test_central_longitude(lon):
        moll = ccrs.Mollweide(central_longitude=lon)
        other_args = {'a=6378137.0', f'lon_0={lon}'}
        check_proj_params('moll', moll, other_args)
    
        assert_almost_equal(np.array(moll.x_limits),
                            [-18040095.6961473, 18040095.6961473],
                            decimal=5)
>       assert_almost_equal(np.array(moll.y_limits),
                            [-9020047.8480736, 9020047.8480736])
E       AssertionError: 
E       Arrays are not almost equal to 7 decimals
E       
E       Mismatched elements: 2 / 2 (100%)
E       Max absolute difference: 0.00017584
E       Max relative difference: 1.94946876e-11
E        x: array([-9020047.8478978,  9020047.8478978])
E        y: array([-9020047.8480736,  9020047.8480736])

cartopy/tests/crs/test_mollweide.py:89: AssertionError
_________________________ test_central_longitude[10.0] _________________________

lon = 10.0

    @pytest.mark.parametrize('lon', [-10.0, 10.0])
    def test_central_longitude(lon):
        moll = ccrs.Mollweide(central_longitude=lon)
        other_args = {'a=6378137.0', f'lon_0={lon}'}
        check_proj_params('moll', moll, other_args)
    
        assert_almost_equal(np.array(moll.x_limits),
                            [-18040095.6961473, 18040095.6961473],
                            decimal=5)
>       assert_almost_equal(np.array(moll.y_limits),
                            [-9020047.8480736, 9020047.8480736])
E       AssertionError: 
E       Arrays are not almost equal to 7 decimals
E       
E       Mismatched elements: 2 / 2 (100%)
E       Max absolute difference: 0.00017584
E       Max relative difference: 1.94946876e-11
E        x: array([-9020047.8478978,  9020047.8478978])
E        y: array([-9020047.8480736,  9020047.8480736])

cartopy/tests/crs/test_mollweide.py:89: AssertionError
=============================== warnings summary ===============================
.pybuild/cpython3_3.10_cartopy/build/cartopy/tests/test_crs.py::TestCRS::test_osgb[True]
.pybuild/cpython3_3.10_cartopy/build/cartopy/tests/test_crs.py::TestCRS::test_osgb[False]
.pybuild/cpython3_3.10_cartopy/build/cartopy/tests/test_crs.py::TestCRS::test_epsg
  /build/python-cartopy-0.20.2+dfsg/.pybuild/cpython3_3.10_cartopy/build/cartopy/tests/test_crs.py:61: UserWarning: uk_os_OSTN15_NTv2_OSGBtoETRS.tif is unavailable; testing OSGB at reduced precision
    warnings.warn(f'{grid_name} is unavailable; '

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=========================== short test summary info ============================
FAILED cartopy/tests/crs/test_interrupted_goode_homolosine.py::test_default[land]
FAILED cartopy/tests/crs/test_interrupted_goode_homolosine.py::test_default[ocean]
FAILED cartopy/tests/crs/test_interrupted_goode_homolosine.py::test_central_longitude[land--10.0]
FAILED cartopy/tests/crs/test_interrupted_goode_homolosine.py::test_central_longitude[land-10.0]
FAILED cartopy/tests/crs/test_interrupted_goode_homolosine.py::test_central_longitude[ocean--10.0]
FAILED cartopy/tests/crs/test_interrupted_goode_homolosine.py::test_central_longitude[ocean-10.0]
FAILED cartopy/tests/crs/test_mollweide.py::test_default - AssertionError: 
FAILED cartopy/tests/crs/test_mollweide.py::test_ellipse_globe - AssertionErr...
FAILED cartopy/tests/crs/test_mollweide.py::test_central_longitude[-10.0] - A...
FAILED cartopy/tests/crs/test_mollweide.py::test_central_longitude[10.0] - As...
= 10 failed, 533 passed, 3 skipped, 121 deselected, 2 xfailed, 2 xpassed, 3 warnings in 27.66s =
```

#### Code to reproduce

Build cartopy with PROJ 9.0.1RC2.

The following changes work around the test failures:
```diff
--- a/lib/cartopy/tests/crs/test_interrupted_goode_homolosine.py
+++ b/lib/cartopy/tests/crs/test_interrupted_goode_homolosine.py
@@ -16,6 +16,7 @@ import cartopy.crs as ccrs
 from .helpers import check_proj_params
 
 
+@pytest.mark.xfail(reason='Fails with PROJ 9.0.1', strict=False)
 @pytest.mark.parametrize("emphasis", ["land", "ocean"])
 def test_default(emphasis):
     igh = ccrs.InterruptedGoodeHomolosine(emphasis=emphasis)
@@ -46,6 +47,7 @@ def test_eccentric_globe(emphasis):
     assert_almost_equal(np.array(igh.y_limits), [-1361.410035, 1361.410035])
 
 
+@pytest.mark.xfail(reason='Fails with PROJ 9.0.1', strict=False)
 @pytest.mark.parametrize(
     ("emphasis", "lon"),
     [("land", -10.0), ("land", 10.0), ("ocean", -10.0), ("ocean", 10.0)],
--- a/lib/cartopy/tests/crs/test_mollweide.py
+++ b/lib/cartopy/tests/crs/test_mollweide.py
@@ -16,6 +16,7 @@ import cartopy.crs as ccrs
 from .helpers import check_proj_params
 
 
+@pytest.mark.xfail(reason='Fails with PROJ 9.0.1', strict=False)
 def test_default():
     moll = ccrs.Mollweide()
     other_args = {'a=6378137.0', 'lon_0=0'}
@@ -37,6 +38,7 @@ def test_sphere_globe():
     assert_almost_equal(moll.y_limits, [-1414.2135624, 1414.2135624])
 
 
+@pytest.mark.xfail(reason='Fails with PROJ 9.0.1', strict=False)
 def test_ellipse_globe():
     globe = ccrs.Globe(ellipse='WGS84')
     with pytest.warns(UserWarning,
@@ -77,6 +79,7 @@ def test_offset():
     assert tuple(np.array(crs.y_limits) - 4321) == crs_offset.y_limits
 
 
+@pytest.mark.xfail(reason='Fails with PROJ 9.0.1', strict=False)
 @pytest.mark.parametrize('lon', [-10.0, 10.0])
 def test_central_longitude(lon):
     moll = ccrs.Mollweide(central_longitude=lon)
```

--- a/lib/cartopy/tests/crs/test_interrupted_goode_homolosine.py
+++ b/lib/cartopy/tests/crs/test_interrupted_goode_homolosine.py
@@ -9,7 +9,7 @@ Tests for the InterruptedGoodeHomolosine coordinate system.
 """
 
 import numpy as np
-from numpy.testing import assert_almost_equal
+from numpy.testing import assert_allclose
 import pytest
 
 import cartopy.crs as ccrs
@@ -24,10 +24,10 @@ def test_default(emphasis):
         check_proj_params("igh", igh, other_args)
     elif emphasis == "ocean":
         check_proj_params("igh_o", igh, other_args)
-    assert_almost_equal(
+    assert_allclose(
         np.array(igh.x_limits), [-20037508.3427892, 20037508.3427892]
     )
-    assert_almost_equal(
+    assert_allclose(
         np.array(igh.y_limits), [-8683259.7164347, 8683259.7164347]
     )
 
@@ -42,8 +42,8 @@ def test_eccentric_globe(emphasis):
     elif emphasis == "ocean":
         check_proj_params("igh_o", igh, other_args)
 
-    assert_almost_equal(np.array(igh.x_limits), [-3141.5926536, 3141.5926536])
-    assert_almost_equal(np.array(igh.y_limits), [-1361.410035, 1361.410035])
+    assert_allclose(np.array(igh.x_limits), [-3141.5926536, 3141.5926536])
+    assert_allclose(np.array(igh.y_limits), [-1361.410035, 1361.410035])
 
 
 @pytest.mark.parametrize(
@@ -60,11 +60,10 @@ def test_central_longitude(emphasis, lon):
     elif emphasis == "ocean":
         check_proj_params("igh_o", igh, other_args)
 
-    assert_almost_equal(
+    assert_allclose(
         np.array(igh.x_limits),
         [-20037508.3427892, 20037508.3427892],
-        decimal=5,
     )
-    assert_almost_equal(
+    assert_allclose(
         np.array(igh.y_limits), [-8683259.7164347, 8683259.7164347]
     )

--- a/lib/cartopy/tests/crs/test_mollweide.py
+++ b/lib/cartopy/tests/crs/test_mollweide.py
@@ -9,7 +9,7 @@ Tests for the Mollweide coordinate system.
 """
 
 import numpy as np
-from numpy.testing import assert_almost_equal
+from numpy.testing import assert_allclose
 import pytest
 
 import cartopy.crs as ccrs
@@ -21,10 +21,10 @@ def test_default():
     other_args = {'a=6378137.0', 'lon_0=0'}
     check_proj_params('moll', moll, other_args)
 
-    assert_almost_equal(np.array(moll.x_limits),
-                        [-18040095.6961473, 18040095.6961473])
-    assert_almost_equal(np.array(moll.y_limits),
-                        [-9020047.8480736, 9020047.8480736])
+    assert_allclose(np.array(moll.x_limits),
+                    [-18040095.6961473, 18040095.6961473])
+    assert_allclose(np.array(moll.y_limits),
+                    [-9020047.8480736, 9020047.8480736])
 
 
 def test_sphere_globe():
@@ -33,8 +33,8 @@ def test_sphere_globe():
     other_args = {'a=1000', 'lon_0=0'}
     check_proj_params('moll', moll, other_args)
 
-    assert_almost_equal(moll.x_limits, [-2828.4271247, 2828.4271247])
-    assert_almost_equal(moll.y_limits, [-1414.2135624, 1414.2135624])
+    assert_allclose(moll.x_limits, [-2828.4271247, 2828.4271247])
+    assert_allclose(moll.y_limits, [-1414.2135624, 1414.2135624])
 
 
 def test_ellipse_globe():
@@ -48,8 +48,8 @@ def test_ellipse_globe():
     check_proj_params('moll', moll, other_args)
 
     # Limits are the same as default since ellipses are not supported.
-    assert_almost_equal(moll.x_limits, [-18040095.6961473, 18040095.6961473])
-    assert_almost_equal(moll.y_limits, [-9020047.8480736, 9020047.8480736])
+    assert_allclose(moll.x_limits, [-18040095.6961473, 18040095.6961473])
+    assert_allclose(moll.y_limits, [-9020047.8480736, 9020047.8480736])
 
 
 def test_eccentric_globe():
@@ -64,8 +64,8 @@ def test_eccentric_globe():
     check_proj_params('moll', moll, other_args)
 
     # Limits are the same as spheres since ellipses are not supported.
-    assert_almost_equal(moll.x_limits, [-2828.4271247, 2828.4271247])
-    assert_almost_equal(moll.y_limits, [-1414.2135624, 1414.2135624])
+    assert_allclose(moll.x_limits, [-2828.4271247, 2828.4271247])
+    assert_allclose(moll.y_limits, [-1414.2135624, 1414.2135624])
 
 
 def test_offset():
@@ -83,11 +83,10 @@ def test_central_longitude(lon):
     other_args = {'a=6378137.0', f'lon_0={lon}'}
     check_proj_params('moll', moll, other_args)
 
-    assert_almost_equal(np.array(moll.x_limits),
-                        [-18040095.6961473, 18040095.6961473],
-                        decimal=5)
-    assert_almost_equal(np.array(moll.y_limits),
-                        [-9020047.8480736, 9020047.8480736])
+    assert_allclose(np.array(moll.x_limits),
+                    [-18040095.6961473, 18040095.6961473])
+    assert_allclose(np.array(moll.y_limits),
+                    [-9020047.8480736, 9020047.8480736])
 
 
 def test_grid():
@@ -100,10 +99,8 @@ def test_grid():
     other_args = {'a=0.7071067811865476', 'b=0.7071067811865476', 'lon_0=0'}
     check_proj_params('moll', moll, other_args)
 
-    assert_almost_equal(np.array(moll.x_limits),
-                        [-2, 2])
-    assert_almost_equal(np.array(moll.y_limits),
-                        [-1, 1])
+    assert_allclose(np.array(moll.x_limits), [-2, 2])
+    assert_allclose(np.array(moll.y_limits), [-1, 1])
 
     lats = np.arange(0, 91, 5)[::-1]
     lons = np.full_like(lats, 90)
@@ -114,14 +111,14 @@ def test_grid():
         0.75894, 0.80591, 0.84739, 0.88362, 0.91477, 0.94096, 0.96229, 0.97882,
         0.99060, 0.99765, 1.00000,
     ])
-    assert_almost_equal(result[:, 0], expected_x, decimal=5)
+    assert_allclose(result[:, 0], expected_x, atol=1e-5)
 
     expected_y = np.array([
         1.00000, 0.97837, 0.94539, 0.90606, 0.86191, 0.81382, 0.76239, 0.70804,
         0.65116, 0.59204, 0.53097, 0.46820, 0.40397, 0.33850, 0.27201, 0.20472,
         0.13681, 0.06851, 0.00000,
     ])
-    assert_almost_equal(result[:, 1], expected_y, decimal=5)
+    assert_allclose(result[:, 1], expected_y, atol=1e-5)
 
 
 def test_sphere_transform():
@@ -135,14 +132,13 @@ def test_sphere_transform():
     other_args = {'a=1.0', 'b=1.0', 'lon_0=-90.0'}
     check_proj_params('moll', moll, other_args)
 
-    assert_almost_equal(np.array(moll.x_limits),
-                        [-2.8284271247461903, 2.8284271247461903],
-                        decimal=2)
-    assert_almost_equal(np.array(moll.y_limits),
-                        [-1.4142135623730951, 1.4142135623730951])
+    assert_allclose(np.array(moll.x_limits),
+                    [-2.8284271247461903, 2.8284271247461903])
+    assert_allclose(np.array(moll.y_limits),
+                    [-1.4142135623730951, 1.4142135623730951])
 
     result = moll.transform_point(-75.0, -50.0, geodetic)
-    assert_almost_equal(result, [0.1788845, -0.9208758])
+    assert_allclose(result, [0.1788845, -0.9208758])
 
     inverse_result = geodetic.transform_point(result[0], result[1], moll)
-    assert_almost_equal(inverse_result, [-75.0, -50.0])
+    assert_allclose(inverse_result, [-75.0, -50.0])

